apps/root/src/pages/transfer/transfer-container/components/recipient-address/index.tsx

```diff
@@ -0,0 +1,122 @@
+import React from 'react';
+import { useAppDispatch } from '@hooks/state';
+import useActiveWallet from '@hooks/useActiveWallet';
+import useReplaceHistory from '@hooks/useReplaceHistory';
+import useCurrentNetwork from '@hooks/useCurrentNetwork';
+import { setRecipient } from '@state/transfer/actions';
+import { useTransferState } from '@state/transfer/hooks';
+import { defineMessage, useIntl } from 'react-intl';
+import { DescriptionOutlinedIcon, IconButton, InputAdornment, TextField } from 'ui-library';
+import { validateAddress } from '@common/utils/parsing';
+import ContactListModal from './components/contact-list-modal';
+
+interface RecipientAddressProps {
+  recipientParam?: string;
+}
+
+const RecipientAddress = ({ recipientParam }: RecipientAddressProps) => {
+  const intl = useIntl();
+  const dispatch = useAppDispatch();
+  const replaceHistory = useReplaceHistory();
+  const activeWallet = useActiveWallet();
+  const [recipientInput, setRecipientInput] = React.useState<string>('');
+  const { token, recipient: storedRecipient } = useTransferState();
+  const currentNetwork = useCurrentNetwork();
+  const [shouldShowContactList, setShouldShowContactList] = React.useState<boolean>(false);
+
+  const inputRegex = RegExp(/^[A-Fa-f0-9x]*$/);
+  const { isValidRecipient } = validateAddress(recipientInput, activeWallet?.address);
+
+  React.useEffect(() => {
+    const { isValidRecipient: isValid } = validateAddress(recipientParam || '', activeWallet?.address);
+    if (recipientParam && isValid) {
+      setRecipientInput(recipientParam);
+      dispatch(setRecipient(recipientParam));
+    }
+  }, [dispatch, validateAddress]);
+
+  const onRecipientChange = (nextValue: string) => {
+    if (!inputRegex.test(nextValue)) {
+      return;
+    }
+    setRecipientInput(nextValue);
+    const { isValidRecipient: isValid } = validateAddress(nextValue, activeWallet?.address);
+
+    if (isValid) {
+      dispatch(setRecipient(nextValue));
+      if (token) {
+        replaceHistory(`/transfer/${currentNetwork.chainId}/${token.address}/${nextValue}`);
+      }
+    } else if (storedRecipient) {
+      // Clear the stored recipient if the input is not valid
+      dispatch(setRecipient(''));
```

Comment:
wait if I put an invalid character all the address gets removed?

Reply:
I tried managing the component state via Redux, but it caused a delay in value updates, impacting the user experience negatively. Now, I only dispatch the value to Redux when it's valid, ensuring the submit method only has a valid or empty value. If Redux has a stored (aways valid) value and the input changes to an invalid value, the stored recipient is reset.

Reply:
gotcha